wallet/client.go

```diff
@@ -39,14 +43,28 @@ func (c *grpcClient) SetServerAddrs(servers []string) {
 	c.servers = servers
 }
 
+func (c *grpcClient) SetTimeoutConnection(timeout time.Duration) {
+	if timeout > 0 {
+		c.timeout = timeout
+	}
+}
+
 func (c *grpcClient) connect() error {
 	if c.conn != nil {
 		return nil
 	}
 
 	for _, server := range c.servers {
-		conn, err := grpc.NewClient(server,
-			grpc.WithTransportCredentials(insecure.NewCredentials()))
+		opts := make([]grpc.DialOption, 0)
+		opts = append(opts, grpc.WithTransportCredentials(insecure.NewCredentials()))
+
+		if len(c.servers) > 1 {
```

Comment:
There is one concern here:
1- User specify a server and timeout at the same time. This condition will not apply the timeout.
2- If `WalletManager`  creates wallet, it try to connect to `localhost`. In this case we may give large number for timeout.